src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,34 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38565_Serialize()
+        {
+            var value = new Regression38565_Parent()
+            {
+                Test = "value1",
+                Child = new Regression38565_Child()
+            };
+
+            var actual = JsonSerializer.ToString(value, new JsonSerializerOptions { IgnoreNullValues = true });
+            Assert.Equal("{\"Test\":\"value1\",\"Child\":{}}", actual);
+        }
+
+        [Fact]
+        public static void Regression38565_Deserialize()
+        {
+            // The json contains nulls to ensure that the dictionary is properly closed
+
+            var json = "{\"Test\":\"value1\",\"Dict\":null,\"Child\":{\"Dict\":null}}";
+            var actual = JsonSerializer.Parse<Regression38565_Parent>(json);
```

Comment:
Can you also add the test case which contains the output from `Regression38565_Serialize`), i.e. we don't have any null values to ignore (but have `IgnoreNullValues = true`)?

Also, nit: avoid var.

```suggestion
            Regression38565_Parent actual = JsonSerializer.Parse<Regression38565_Parent>(json);
```

Reply:
> Can you also add the test case which contains the output from

Fixed.

Reply:
> Also, nit: avoid var.

Also, fixed. But I think this would be apparent to the majority of developers. If your `Parse<T>` method doesn't return `T`, you probably have bigger problems.